### PR TITLE
Update to PHP-Parser v4

### PR DESCRIPTION
**Pull request by carusogabriel**
PHP-Parser v4 [was released](https://github.com/nikic/PHP-Parser/commit/9c18e3db49fa469f5feffe40dbd7b1ec2b61e41f), and now we can remove our own fork of `Roave/BetterReflection`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,10 +11,10 @@
         "jean85/pretty-package-versions": "^1.1",
         "nette/robot-loader": "^3.0",
         "nette/utils": "^2.5",
-        "nikic/php-parser": "4.0.x-dev",
+        "nikic/php-parser": "^4.0",
         "phpdocumentor/reflection-docblock": "^4.3",
         "phpdocumentor/type-resolver": "^0.4",
-        "rector/better-reflection": "^3.0.6",
+        "roave/better-reflection": "dev-master",
```
Comment (carusogabriel) [inline composer.json]: Waiting for https://github.com/Roave/BetterReflection/pull/409#issuecomment-369405816

Comment (TomasVotruba) [reply composer.json]: Such PRs could be added here via https://blog.martinhujer.cz/17-tips-for-using-composer-efficiently/#comment-3697775212

**Comment by TomasVotruba** [comment]
Horray! Party time :) I'll get to this tomorrow

File: composer.json (hunk 1 above)
Comment (carusogabriel) [reply composer.json]: @TomasVotruba Thanks, didn't know that one

File: composer.json [hunk 2]
```diff
@@ -12,10 +12,10 @@
         "jean85/pretty-package-versions": "^1.1",
         "nette/robot-loader": "^3.0",
         "nette/utils": "^2.5",
-        "nikic/php-parser": "4.0.x-dev",
+        "nikic/php-parser": "^4.0",
         "phpdocumentor/reflection-docblock": "^4.3",
         "phpdocumentor/type-resolver": "^0.4",
-        "rector/better-reflection": "^3.0.6",
+        "roave/better-reflection": "dev-master#dd2c2d0",
```
Comment (TomasVotruba) [inline composer.json]: To really use stable dependencies, here use `rector/better-reflection` and update it first to `nikic/php-parser` 4.0.
I'll release new tag for it then

**Comment by TomasVotruba** [comment]
Thank you Gabriel

**Comment by carusogabriel** [comment]
I'll track https://github.com/Roave/BetterReflection/pull/410, so when that gets merge, we can remove our fork :+1:

**Comment by TomasVotruba** [comment]
Seeing the stuck discussion - though I suggested working solution - it might take many months. 
We don't need that really :)

**Pull request merged**